### ISSUE TITLE
TypeScript warnings with MMKVWrapper

### ISSUE THREAD

**Issue by aloukissas** [apollographql/apollo-cache-persist#460]
It looks like that the `MMKVInterface` from `react-native-mmkv` and that from this project differ in some return types (see below).

tl;dr: the `getBoolean` and `getNumber` functions can return `undefined` in `react-native-mmkv`, which is where TypeScript has an issue; ideally, `apollo-cache-persist` consumes the `MMKVInterface` directly from `react-native-mmkv` instead of redefining it.

## `react-native-mmkv` version

```typescript
interface MMKVInterface {
    /**
     * Set a value for the given `key`.
     */
    set: (key: string, value: boolean | string | number) => void;
    /**
     * Get the boolean value for the given `key`, or `undefined` if it does not exist.
     *
     * @default undefined
     */
    getBoolean: (key: string) => boolean | undefined;
    /**
     * Get the string value for the given `key`, or `undefined` if it does not exist.
     *
     * @default undefined
     */
    getString: (key: string) => string | undefined;
    /**
     * Get the number value for the given `key`, or `undefined` if it does not exist.
     *
     * @default undefined
     */
    getNumber: (key: string) => number | undefined;
    /**
     * Checks whether the given `key` is being stored in this MMKV instance.
     */
    contains: (key: string) => boolean;
    /**
     * Delete the given `key`.
     */
    delete: (key: string) => void;
    /**
     * Get all keys.
     *
     * @default []
     */
    getAllKeys: () => string[];
    /**
     * Delete all keys.
     */
    clearAll: () => void;
    /**
     * Sets (or updates) the encryption-key to encrypt all data in this MMKV instance with.
     *
     * To remove encryption, pass `undefined` as a key.
     *
     * Encryption keys can have a maximum length of 16 bytes.
     */
    recrypt: (key: string | undefined) => void;
    /**
     * Adds a value changed listener. The Listener will be called whenever any value
     * in this storage instance changes (set or delete).
     *
     * To unsubscribe from value changes, call `remove()` on the Listener.
     */
    addOnValueChangedListener: (onValueChanged: (key: string) => void) => Listener;
}
```

## `apollo-cache-persist` version

```typescript
interface MMKVInterface {
    set: (key: string, value: boolean | string | number) => void;
    getBoolean: (key: string) => boolean;
    getString: (key: string) => string | undefined;
    getNumber: (key: string) => number;
    delete: (key: string) => void;
    getAllKeys: () => string[];
    clearAll: () => void;
}
```

**Comment by wodCZ** [comment]
Hi @aloukissas, thanks for bringing this up!

The incompatibility you're mentioning was introduced in recent [react-native-mmkv release](https://github.com/mrousavy/react-native-mmkv/releases/tag/v2.3.0).

We've previously discussed how to import interfaces directly from the relevant libraries, but we haven't found a clear way to do that without affecting the resulting bundles. That's why we've decided to "snapshot" the interfaces, even though that means that new releases of the libraries might introduce incompatibilities such as this. More on that [here](https://github.com/apollographql/apollo-cache-persist/pull/391#issuecomment-716162164).

**What now?** We'll need to update the local interface to match the new version. I'd like to keep the local wrapper compatible with previous versions too if possible. This library actually doesn't require neither `getBoolean` nor `getNumber`, so I think we can safely remove them from our interface — that should be compatible with both.

I'm currently a bit swamped with work, @aloukissas would you mind creating a PR?

**Comment by aloukissas** [comment]
NP, I'll try to get a PR out :)

**Comment by wodCZ** [comment]
Should be fixed when #462 is released.

**Comment by paulrostorp** [comment]
@wodCZ Any idea when this will be released ?

**Comment by wodCZ** [comment]
I'll push the new release on Monday. You should be able to use `https://github.com/apollographql/apollo-cache-persist#16641d87a1604b935308b005bb3697455c0f464b` until then.

```sh
yarn add https://github.com/apollographql/apollo-cache-persist#16641d87a1604b935308b005bb3697455c0f464b
```

**Comment by wodCZ** [comment]
Released in 0.14.1.
